app/models/flat_reasons_for_rejection_extract.rb

```diff
@@ -0,0 +1,183 @@
+  end
+
+  def plagiarism?
+    sub_level_reasons('honesty_and_professionalism_concerns', 'plagiarism')
+  end
+
+  def references?
+    sub_level_reasons('honesty_and_professionalism_concerns', 'references')
+  end
+
+  def honesty_and_professionalism_concerns_information_false_or_inaccurate_details
+    other_detail_or_what_to_improve('honesty_and_professionalism_concerns_information_false_or_inaccurate_details')
+  end
+
+  def honesty_and_professionalism_concerns_plagiarism_details
+    other_detail_or_what_to_improve('honesty_and_professionalism_concerns_plagiarism_details')
+  end
+
+  def honesty_and_professionalism_concerns_references_details
+    other_detail_or_what_to_improve('honesty_and_professionalism_concerns_references_details')
+  end
+
+  def honesty_and_professionalism_concerns_other_details
+    other_detail_or_what_to_improve('honesty_and_professionalism_concerns_other_details')
+  end
+
+  def safeguarding?
+    top_level_reasons('safeguarding_y_n')
+  end
+
+  def candidate_disclosed_information?
+    sub_level_reasons('safeguarding_concerns', 'candidate_disclosed_information')
+  end
+
+  def safeguarding_concerns_candidate_disclosed_information_details
+    other_detail_or_what_to_improve('safeguarding_concerns_candidate_disclosed_information_details')
+  end
+
+  def vetting_disclosed_information?
+    sub_level_reasons('safeguarding_concerns', 'vetting_disclosed_information')
+  end
+
+  def safeguarding_concerns_vetting_disclosed_information_details
+    other_detail_or_what_to_improve('safeguarding_concerns_vetting_disclosed_information_details')
+  end
+
+  def safeguarding_concerns_other_details
+    other_detail_or_what_to_improve('safeguarding_concerns_other_details')
+  end
+
+  # These three methods are copied from the application_choices_export.rb .... currently doesn't provide enough granularity
+  def format_structured_rejection_reasons
+    return nil if @structured_rejection_reasons.blank?
+
+    select_high_level_rejection_reasons(@structured_rejection_reasons)
+    .keys
+    .map { |reason| format_reason(reason) }
+    .join("\n")
+  end
+
+  def select_high_level_rejection_reasons(structured_rejection_reasons)
+    structured_rejection_reasons.select { |reason, value| value == 'Yes' && reason.include?('_y_n') }
+  end
+
+  def format_reason(reason)
+    reason
+    .delete_suffix('_y_n')
+    .humanize
+  end
+
+  private
+
+  def top_level_reasons(key)
+    return nil if @structured_rejection_reasons["#{key}"].blank?
+
+    @structured_rejection_reasons.select { |reason, value| value == 'Yes' && reason == "#{key}" }.present?
+  end
+
+  def sub_level_reasons(key, value)
+    return nil if @structured_rejection_reasons["#{key}"].blank?
+
+    @structured_rejection_reasons["#{key}"].include?("#{value}")
+  end
+
+  def other_detail_or_what_to_improve(key)
+    return nil if @structured_rejection_reasons["#{key}"].blank?
+
+    @structured_rejection_reasons["#{key}"]
+  end
+end
```

Comment:
I think this belongs in a presenter and not a model, similar to the [RejectedApplicationChoicePresenter](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/app/presenters/rejected_application_choice_presenter.rb)